cpp/src/arrow/compute/kernels/vector_selection_test.cc

```diff
@@ -2365,38 +2365,91 @@ TEST_F(TestDropNullKernelWithTable, DropNullTableWithSlices) {
   });
 }
 
-TEST(TestIndicesNonZero, IndicesNonZero) {
+template <typename Type>
+class TestIndicesNonZero : public ::testing::Test {};
+
+TYPED_TEST_SUITE(TestIndicesNonZero, NumericArrowTypes);
+TYPED_TEST(TestIndicesNonZero, IndicesNonZero) {
   Datum actual;
   std::shared_ptr<Array> result;
+  auto type = TypeTraits<TypeParam>::type_singleton();
 
-  ASSERT_OK_AND_ASSIGN(
-      actual,
-      CallFunction("indices_nonzero", {ArrayFromJSON(uint32(), "[null, 50, 0, 10]")}));
+  ASSERT_OK_AND_ASSIGN(actual, CallFunction("indices_nonzero",
+                                            {ArrayFromJSON(type, "[null, 50, 0, 10]")}));
   result = actual.make_array();
-  AssertArraysEqual(*result, *ArrayFromJSON(uint64(), "[1, 3]"));
+  AssertArraysEqual(*ArrayFromJSON(uint64(), "[1, 3]"), *result);
+
+  // empty
+  ASSERT_OK_AND_ASSIGN(actual,
+                       CallFunction("indices_nonzero", {ArrayFromJSON(type, "[]")}));
+  result = actual.make_array();
+  AssertArraysEqual(*ArrayFromJSON(uint64(), "[]"), *result);
+
+  // chunked
+  ChunkedArray chunked_arr(
+      {ArrayFromJSON(type, "[1, 0, 3]"), ArrayFromJSON(type, "[4, 0, 6]")});
+  ASSERT_OK_AND_ASSIGN(
+      actual, CallFunction("indices_nonzero", {static_cast<Datum>(chunked_arr)}));
```

Comment:
I think `{chunked_arr}` would simply work and it would convert to a Datum implicitly.

Reply:
For me, I get a compiler error: `error: no matching function for call to 'CallFunction' actual, CallFunction("indices_nonzero", {chunked_arr}));`